python/ray/function_manager.py

```diff
@@ -121,22 +120,20 @@ def from_function(cls, function):
         function_name = function.__name__
         class_name = ""
 
-        function_source_hasher = hashlib.sha1()
+        pickled_function_hasher = hashlib.sha1()
         try:
             # If we are running a script or are in IPython, include the source
             # code in the hash.
-            source = inspect.getsource(function)
-            if sys.version_info[0] >= 3:
-                source = source.encode()
-            function_source_hasher.update(source)
-            function_source_hash = function_source_hasher.digest()
+            pickled_function = pickle.dumps(function)
```

Comment:
@edoakes are you ok with this change?

Reply:
As discussed, it's inefficient because if the same function is defined on many workers, they will all be exported/imported separately. However, it should be correct in all cases.

Reply:
If we're going to do this, let's at least only call `pickle.dumps` once. Also, this could be really bad for a large cluster where many workers are submitting/executing the same tasks - can you run a benchmark to test it out?